shared/validation/languages/common.js

```diff
@@ -0,0 +1,36 @@
+var unicode = require('unicode-properties');
+var decode = require('urldecode');
+var striptags = require('striptags');
+
+function _strip_string(sentence) {
+    //letters
+    var allowed_categories = ["LC", "Ll", "Lm", "Lo", "Lt", "Lu"];
+    //numbers
+    allowed_categories.concat(["Nd", "Nl", "No"]);
+    //marks
+    allowed_categories.concat(["Mc", "Me", "Mn"]);
+    //punctuation
+    allowed_categories.concat(["Pc", "Pd", "Pe", "Pf", "Pi", "Po", "Ps"]);
+    //symbol
+    allowed_categories.concat(["Sc", "Sk", "Sm", "So"]);
+    //space
+    allowed_categories.concat(["Zs"]);
+    let st = "";
+    for (var s of sentence) {
+        if (allowed_categories.includes(unicode.getCategory(s.charCodeAt()))) {
+            st = st.concat(s);
+        }
+    }
+    return (st);
+}
+export function Common (sentence) {
+    // Decode any URL encoded elements of sentence
+    sentence = decode(sentence);
+    // Remove any HTML tags
+    sentence = striptags(sentence);
```

Comment:
where did you see html tags? These should not even be allowed and sentences should be marked as invalid before we even come here.